Cargo.toml

```diff
@@ -12,4 +12,9 @@ keywords = ["interval", "tree", "bound", "exclusive", "difference"]
 categories = ["algorithms", "data-structures"]
 
 [dependencies]
-rand = "0.8"
\ No newline at end of file
+rand = "0.8"
+serde = { version = "1.0.145", optional = true , features = ["derive"] }
```

Comment:
Is there a reason why we have "1.0.145" specifically? Otherwise, I would prefer to just have "1.0", so that we do not limit to one patch number -- just like we do for `rand` above.